### PR TITLE
ENG-17382: Remove legacycompile from test_apps run.sh

### PR DESCRIPTION
**Pull request by johnTcrawford**
On the 'voltdb' community side, modified run.sh files, replacing the
srccompile & catalog functions with jars & jars-if-needed, and updating
server & help functions accordingly (& other minor changes, as needed),
for the following 'test_apps', which previously used the long-deprecated
'voltdb legacycompile' (in the catalog function): aggbench,
csvbenchmark, kvbenchmark, matviewbenchmark, measure-overhead,
oneshotkv, scans, schemachange, voter-selfcheck (also needed a tweak in
deployment-secure.xml), and ycsb-zipfian. Also, deleted the following
obsolete 'test_apps': eng1969, overhead, voter-daemons, which are no
longer used (they have have no references on the 'pro' side, including
in workloads.py.

### CODE REVIEW

File: tests/test_apps/csvbenchmark/run.sh [hunk 1]
```diff
@@ -37,51 +37,61 @@ LICENSE="$VOLTDB_VOLTDB/license.xml"
...
 function clean() {
-    rm -rf obj debugoutput $APPNAME.jar voltdbroot voltdbroot
+    rm -rf obj debugoutput ${APPNAME}.jar voltdbroot voltdbroot
 }
 
-# compile the source code for procedures and the client
-function srccompile() {
-    return
+# compile the source code for procedures into jarfiles
+function jars() {
```
Comment (jwiebema1) [inline tests/test_apps/csvbenchmark/run.sh]: This is failing in the performance tests.  http://ci:8080/job/branch-performance-140-csv-ENG-17214-app-legacycompile2/1/

File: tests/test_apps/scans/run.sh [hunk 2]
```diff
@@ -32,33 +32,46 @@ HOST="localhost"
...
 # remove build artifacts
 function clean() {
-    rm -rf obj debugoutput $APPNAME.jar voltdbroot voltdbroot
+    rm -rf obj debugoutput ${APPNAME}.jar voltdbroot voltdbroot
 }
 
-# compile the source code for procedures and the client
-function srccompile() {
+# compile the source code for procedures and the client into jarfiles
+function jars() {
```
Comment (jwiebema1) [inline tests/test_apps/scans/run.sh]: This is failing in the performance tests.  http://ci:8080/job/branch-performance-040-scans-ENG-17214-app-legacycompile2/1/console

File: tests/test_apps/oneshotkv/run.sh [hunk 3]
```diff
@@ -35,38 +35,56 @@ HOST="localhost"
...
 # remove build artifacts
 function clean() {
-    rm -rf obj debugoutput $APPNAME.jar voltdbroot voltdbroot
+    rm -rf obj debugoutput ${APPNAME}.jar voltdbroot voltdbroot
 }
 
-# compile the source code for procedures and the client
-function srccompile() {
+# compile the source code for procedures and the client into jarfiles
+function jars() {
```
Comment (jwiebema1) [inline tests/test_apps/oneshotkv/run.sh]: This fails on performance tests:   http://ci:8080/job/branch-performance-080-oneshot-mp-ENG-17214-app-legacycompile2/1/console

File: tests/test_apps/scans/run.sh [hunk 4]
```diff
@@ -68,15 +81,15 @@ function client() {
 
 function benchmark() {
     srccompile
```
Comment (jwiebema1) [inline tests/test_apps/scans/run.sh]: This function doesn't exist anymore, should be jars

File: tests/test_apps/aggbench/run.sh [hunk 5]
```diff
@@ -42,59 +44,60 @@ HOST="localhost"
...
-	for x in `seq 1 5`; do echo "Query $x" ; echo "exec Q${x}"| ${SQLCMD} | grep "rows" ; done
+#   for x in `seq 1 5`; do echo "Query $x" ; echo "exec Q${x}"| ${SQLCMD} | grep "rows" ; done
+    for x in `seq 1 5`; do echo "Query $x" ; echo "exec Q${x}"| ${SQLCMD} | grep "rows" ; done
 }
 
+# run the client that drives the example
 function client() {
-    srccompile
-    java -classpath obj:$APPCLASSPATH:obj -Dlog4j.configuration=file://$LOG4J \
+    #srccompile
```
Comment (jwiebema1) [inline tests/test_apps/aggbench/run.sh]: should this be jars-ifneeded

File: tests/test_apps/measure-overhead/run.sh [hunk 6]
```diff
@@ -32,33 +34,46 @@ HOST="localhost"
...
 # remove build artifacts
 function clean() {
-    rm -rf obj debugoutput $APPNAME.jar voltdbroot voltdbroot
+    rm -rf obj debugoutput ${APPNAME}.jar voltdbroot voltdbroot
 }
 
-# compile the source code for procedures and the client
-function srccompile() {
+# compile the source code for procedures and the client into jarfiles
+function jars() {
```
Comment (jwiebema1) [inline tests/test_apps/measure-overhead/run.sh]: this is failing in the command log tests:  http://ci:8080/job/branch-performance-220-cl-ENG-17214-app-legacycompile2/1/console

File: tests/test_apps/schemachange/run.sh [hunk 7]
```diff
@@ -13,35 +13,48 @@ function clean() {
     rm -rf obj debugoutput $APPNAME.jar voltdbroot
 }
 
-# compile the source code for procedures and the client
-function srccompile() {
+# compile the source code for the client into jarfiles
+function jars() {
```
Comment (jwiebema1) [inline tests/test_apps/schemachange/run.sh]: This is failing in the system tests;  http://volt0/apprunner/ENG-17214-app-legacycompile2/jenkins-branch-system-test-schemachange-ENG-17214-app-legacycompile2-2/schemachange-recover-volt13i-20191215142026-FAIL.tar.gz

**Comment by johnTcrawford** [comment]
Old, superseded by ENG-17214-app-legacycompile4.

**Pull request closed without merging**